File: examples/python/midjourney_example.py
import os
import time
import requests
from dotenv import load_dotenv

# Load environment variables from .env file
load_dotenv()

LEGNEXT_API_KEY = os.getenv("LEGNEXT_API_KEY", "YOUR_LEGNEXT_API_KEY")
LEGNEXT_API_BASE = os.getenv("LEGNEXT_API_BASE", "https://api.legnext.ai/api")


def generate_image(prompt: str) -> dict:
    """
    Generate an image using Midjourney via Legnext API

    Args:
        prompt: Text prompt for image generation

    Returns:
        Job creation response dictionary
    """
    url = f"{LEGNEXT_API_BASE}/v1/diffusion"
    headers = {
        "x-api-key": LEGNEXT_API_KEY,
        "Content-Type": "application/json"
    }
    data = {
        "text": prompt
    }

    response = requests.post(url, headers=headers, json=data)

    if not response.ok:
        raise Exception(f"Failed to generate image: {response.status_code} - {response.text}")

    return response.json()


def get_job_status(job_id: str) -> dict:
    """
    Check the status of a job

    Args:
        job_id: Job ID to check

    Returns:
        Job status response dictionary
    """
    url = f"{LEGNEXT_API_BASE}/v1/job/{job_id}"
    headers = {
        "x-api-key": LEGNEXT_API_KEY
    }

    response = requests.get(url, headers=headers)

    if not response.ok:
        raise Exception(f"Failed to get job status: {response.status_code} - {response.text}")

    return response.json()


def wait_for_completion(job_id: str, max_attempts: int = 60, interval_seconds: int = 5) -> dict:
    """
    Wait for job completion by polling

    Args:
        job_id: Job ID to wait for
        max_attempts: Maximum number of polling attempts (default: 60)
        interval_seconds: Interval between polls in seconds (default: 5)

    Returns:
        Completed job details dictionary
    """
    for i in range(max_attempts):
        job = get_job_status(job_id)

        print(f"[{i + 1}/{max_attempts}] Job status: {job['status']}")

        if job["status"] == "completed":
            return job
        elif job["status"] == "failed":
            error = job.get("error", "Unknown error")
            raise Exception(f"Job failed: {error}")

        # Wait before next poll
        time.sleep(interval_seconds)

    raise Exception("Job did not complete within the maximum wait time")


def main():
    try:
        print("Starting Midjourney image generation via Legnext API...\n")

        prompt = "A beautiful sunset over the snow mountains --v 7"
        print(f"Prompt: {prompt}\n")

        # Step 1: Generate image
        print("Step 1: Submitting image generation request...")
        create_response = generate_image(prompt)
        job_id = create_response["job_id"]
        print(f"✓ Job created with ID: {job_id}\n")

        # Step 2: Wait for completion
        print("Step 2: Waiting for image generation to complete...")
        completed_job = wait_for_completion(job_id)

        # Step 3: Display results
        print("\n✓ Image generation completed!\n")
        print("Results:")
        print(f"- Job ID: {completed_job['job_id']}")
        print(f"- Status: {completed_job['status']}")
        print(f"- Model: {completed_job['model']}")

        if completed_job.get("output") and completed_job["output"].get("image_url"):
            print(f"\nGenerated Image URL:")
            print(completed_job["output"]["image_url"])

        if completed_job.get("output") and completed_job["output"].get("actions"):
            print(f"\nAvailable Actions:")
            for action in completed_job["output"]["actions"]:
                print(f"- {action}")

        if completed_job.get("meta"):
            print(f"\nMetadata:")
            print(f"- Created: {completed_job['meta'].get('created_at')}")
            print(f"- Completed: {completed_job['meta'].get('completed_at')}")

    except Exception as error:
        print(f"Error: {error}")
        exit(1)


if __name__ == "__main__":
    main()

File: examples/javascript/midjourney-example.js
import dotenv from 'dotenv';
import fetch from 'node-fetch';

// Load environment variables from .env file
dotenv.config();

const LEGNEXT_API_KEY = process.env.LEGNEXT_API_KEY || "YOUR_LEGNEXT_API_KEY";
const LEGNEXT_API_BASE = process.env.LEGNEXT_API_BASE || "https://api.legnext.ai/api";

/**
 * Generate an image using Midjourney via Legnext API
 * @param {string} prompt - Text prompt for image generation
 * @returns {Promise<object>} - Job creation response
 */
async function generateImage(prompt) {
  const response = await fetch(`${LEGNEXT_API_BASE}/v1/diffusion`, {
    method: 'POST',
    headers: {
      'x-api-key': LEGNEXT_API_KEY,
      'Content-Type': 'application/json'
    },
    body: JSON.stringify({
      text: prompt
    })
  });

  if (!response.ok) {
    const error = await response.text();
    throw new Error(`Failed to generate image: ${response.status} - ${error}`);
  }

  return await response.json();
}

/**
 * Check the status of a job
 * @param {string} jobId - Job ID to check
 * @returns {Promise<object>} - Job status response
 */
async function getJobStatus(jobId) {
  const response = await fetch(`${LEGNEXT_API_BASE}/v1/job/${jobId}`, {
    method: 'GET',
    headers: {
      'x-api-key': LEGNEXT_API_KEY
    }
  });

  if (!response.ok) {
    const error = await response.text();
    throw new Error(`Failed to get job status: ${response.status} - ${error}`);
  }

  return await response.json();
}

/**
 * Wait for job completion by polling
 * @param {string} jobId - Job ID to wait for
 * @param {number} maxAttempts - Maximum number of polling attempts (default: 60)
 * @param {number} intervalMs - Interval between polls in milliseconds (default: 5000)
 * @returns {Promise<object>} - Completed job details
 */
async function waitForCompletion(jobId, maxAttempts = 60, intervalMs = 5000) {
  for (let i = 0; i < maxAttempts; i++) {
    const job = await getJobStatus(jobId);

    console.log(`[${i + 1}/${maxAttempts}] Job status: ${job.status}`);

    if (job.status === 'completed') {
      return job;
    } else if (job.status === 'failed') {
      throw new Error(`Job failed: ${job.error || 'Unknown error'}`);
    }

    // Wait before next poll
    await new Promise(resolve => setTimeout(resolve, intervalMs));
  }

  throw new Error('Job did not complete within the maximum wait time');
}

async function main() {
  try {
    console.log('Starting Midjourney image generation via Legnext API...\n');

    const prompt = "A beautiful sunset over the snow mountains --v 7";
    console.log(`Prompt: ${prompt}\n`);

    // Step 1: Generate image
    console.log('Step 1: Submitting image generation request...');
    const createResponse = await generateImage(prompt);
    const jobId = createResponse.job_id;
    console.log(`✓ Job created with ID: ${jobId}\n`);

    // Step 2: Wait for completion
    console.log('Step 2: Waiting for image generation to complete...');
    const completedJob = await waitForCompletion(jobId);

    // Step 3: Display results
    console.log('\n✓ Image generation completed!\n');
    console.log('Results:');
    console.log(`- Job ID: ${completedJob.job_id}`);
    console.log(`- Status: ${completedJob.status}`);
    console.log(`- Model: ${completedJob.model}`);

    if (completedJob.output && completedJob.output.image_url) {
      console.log(`\nGenerated Image URL:`);
      console.log(completedJob.output.image_url);
    }

    if (completedJob.output && completedJob.output.actions) {
      console.log(`\nAvailable Actions:`);
      completedJob.output.actions.forEach(action => {
        console.log(`- ${action}`);
      });
    }

    if (completedJob.meta) {
      console.log(`\nMetadata:`);
      console.log(`- Created: ${completedJob.meta.created_at}`);
      console.log(`- Completed: ${completedJob.meta.completed_at}`);
    }

  } catch (error) {
    console.error('Error:', error.message);
    process.exit(1);
  }
}

main();
